server/src/services/projectService.ts

```diff
@@ -597,6 +598,64 @@ const setScheduleData = async (schedulePayload: SetSchedulePayload) => {
   return statusPayload;
 }
 
+const editScheduleData = async (payload: SetSchedulePayload) => {
+  try {
+    const scheduleData: ScheduleDataMongoResponse = await ScheduleDataSchema.findOne({"projectID": payload.projectID});
+    const keyData: ScheduleKeys | null = scheduleData.scheduleKeys.get(payload.locationID) ?? null;
+    
+    if (keyData) {
+      const parts = keyData.key.split(" ");
+      const date = parts.slice(0, 3).join(' ');
+      const time = parts.slice(3).join(' ');
+      const validDate = date.replace(/(\d+)(st|nd|rd|th),/, '$1,');
+      const dateUnix = new Date(validDate).getTime()
+
+      const oldPayload = {
+        time: time,
+        date: date,
+        dateUnix: dateUnix,
+        projectID: payload.projectID,
+        locationID: payload.locationID,
+        noteMessage: payload.noteMessage,
+        noteName: payload.noteName,
+        notePriority: payload.notePriority,
+        duration: keyData.duration,
+      }
+
+      const responseDel = await projectService.deleteSchedule(payload.locationID, payload.projectID);
+      //If there are schedule conflicts restore old data and throw error
+      const conflict = await checkIfConflictsExists(payload);
+      if (!conflict) {
+        const responseSet = await projectService.setScheduleData(payload);
+        return responseSet;
+      } else {
+        const responseSet = await projectService.setScheduleData(oldPayload);
+        const statusPayload: {status: StatusPayload} = {
+          status: {
+            statusCode: STATUS_CODES.BadRequest,
+            errorCause: ERROR_CAUSE.Schedule,
+            errorData: ERROR_DATA.ScheduleConflict,
+          }
+        }
+  
+        return statusPayload;
+      }
+    }
```

Comment:
Right now this is kind of buggy. When I edit the schedule to overlap two existing schedules, two errors are displayed on the front-end, the schedule is still set and the original schedule gets deleted on refresh.


I think it'll work if we move the `const responseDel = await projectService.deleteSchedule(payload.locationID, payload.projectID);` into the if statement, before the `setScheduleData`.

With this approach, we dont need the `const responseSet = await projectService.setScheduleData(oldPayload);` in the else statement because the original data is still present.

Reply:
The issue with having the responseDel in the if statement is that we have to delete the existing note before we check for conflicts. I do see that error though so I will try to debug that